container-images/tcib/base/os/tempest/run_tempest.sh

```diff
@@ -18,15 +56,10 @@ if [ ! -z ${USE_EXTERNAL_FILES} ]; then
     cp ${TEMPEST_PATH}clouds.yaml $HOME/.config/openstack/clouds.yaml
 fi
 
-if [ -f ${TEMPEST_PATH}profile.yaml ]; then
```

Comment:
if we delete this block, we'll break the ci-framework usage of this image ... they generate profile.yaml on the fly - https://github.com/openstack-k8s-operators/ci-framework/blob/56d9e69a2ce00d96dcf006f86b088ee944c1a39d/ci_framework/roles/tempest/tasks/configure-tempest.yml#L34 

Then the file would be discovered by this if-statement and passed to tempestconf, if we remove it, the generated profile.yaml file will not make it to tempestconf.

I see couple of options:
1. set TEMPESTCONF_PROFILE somewhere in ci-framework/roles/tempest
2. extend the logic of this if-statement so that if TEMPESTCONF_PROFILE is not set,  ${TEMPEST_PATH}profile.yaml will be passed to --profile if the file exists.

Reply:
That's a good point. 

What do you think about sticking only with TEMPESTCONF_PROFILE and removing the option of passing the discover-tempest-config parameters by mounting the profile.yaml to the image? I would stick with only one interface. Keeping it simple. We are still in the early stages so I think we can afford it if we modify ci-framework correctly (?). Just an idea.

Reply:
I'm afraid I don't follow. As it is right now, there are 2 options how to pass arguments to tempestconf.
1. generate a custom profile.yaml that is mounted to the container - that's how ci-framework sets options for discover-tempest-config right now - https://github.com/openstack-k8s-operators/ci-framework/blob/d776dc8babe7117c2d29ad722b70117da1bde09a/ci_framework/roles/tempest/tasks/configure-tempest.yml#L34
2. set desired `TEMPESTCONF_*` variables

If you remove this if-statement, the profile.yaml file that is generated by ci-framework will be ignored unless we set `TEMPESTCONF_PROFILE` in ci-framework.